stellarphot/gui_tools/seeing_profile_functions.py

```diff
@@ -464,12 +466,16 @@ def __init__(self, imagewidget=None, width=500):
         big_box = ipw.HBox()
         big_box = ipw.GridspecLayout(1, 2)
         layout = ipw.Layout(width='20ch')
-        hb = ipw.HBox()
-        self.ap_t = ipw.IntText(description='Aperture radius', value=5, layout=layout, style=desc_style)
-        self.in_t = ipw.IntText(description='Inner annulus', value=10, layout=layout, style=desc_style)
-        self.out_t = ipw.IntText(description='Outer annulus', value=20, layout=layout, style=desc_style)
+        vb = ipw.VBox()
+        self.aperture_settings_file_name = ipw.Text(
+            description="Aperture settings file name",
+            style={'description_width': 'initial'},
+            value="aperture_settings.json"
+        )
+        self.aperture_settings = ui_generator(ApertureSettings)
+        self.aperture_settings.path = Path(self.aperture_settings_file_name.value)

```

Comment:
Just confirming, this generates a ipywidgets UI based on the ApertureSettings pydantic mode and then sets a path for saying the settings to json (both being parameters set via ipywidgets widgets)?

Reply:
Correct